internal/service/project/resource_project.go

```diff
@@ -603,23 +603,23 @@ type AdditionalProperties struct {
 }
 
 // GetProjectPropsFromAPI fetches properties obtained from complementary endpoints associated with a project.
-func GetProjectPropsFromAPI(ctx context.Context, client GroupProjectService, projectID string) (*AdditionalProperties, error) {
-	teams, _, err := client.ListProjectTeams(ctx, projectID)
+func GetProjectPropsFromAPI(ctx context.Context, client *admin.APIClient, projectID string) (*AdditionalProperties, error) {
+	teams, _, err := client.TeamsApi.ListProjectTeams(ctx, projectID).Execute()
```

Comment:
in the PR @AgustinBettati did, he passed the API itself, WDYT?
I see here you need TeamApi & ProjectApi, but still it's clearer to pass them instead of client so the API needed are specified

Reply:
I don't have a strong opinion here... 
I like that it makes the callers of GetProjectPropsFromAPI as it is called in 6 places.
Therefore, I would prefer to keep it as is. But I can change it if you insist :D

Reply:
@AgustinBettati WDYT?

Reply:
I am more on the side of receiving the specific interfaces that are used just to have more clarity when seeing the function signature. You also have the saying "accept interfaces and return structs"

Reply:
Alright. Thanks for the input :smile:
I have heard that sentence before, and I think the simpler the interface is the better :sweat_smile:
Changed in [b781b71](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2045/commits/b781b71f45f8db1b5555c70847644d57fc30763f)